src/Common/Menus/Menu_Contract.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * The interface for all menu objects.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus;
+
+/**
+ * Interface Menu_Contract
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Menus
+ */
+interface Menu_Contract {
```

Comment:
While I'm all for depending on abstractions and not implementations, if any class that will be "a menu" will end up extending the base abstract class, this interface would be a duplication. An interface will not offer more flexibility when it comes to testing.
If the objective is to guide the development of menu pages and make it easy, then an abstract class will do just fine. The abstract class will define the core mechanics provided the extension checks all the required methods.

Reply:
This was meant as an example - not to be extended.

The aim here was to avoid another `Main.php` with _everything_ in it - the contract enforces the base implementation (and yes, the abstract would too) and I have found it useful so far. Removing it is not out of the question though.

Reply:
Removed the contract interface.